pennylane/math/utils.py

```diff
@@ -229,7 +232,7 @@ def requires_grad(tensor):
     ...     print(requires_grad(x))
     True
     """
-    interface = get_interface(tensor)
+    interface = interface or get_interface(tensor)

```

Comment:
Can't we always just do this? Or do you want to save costs if we already know the interface?

Reply:
Oh, actually this is to enable different behaviour! Sometimes, we want to consider a singular interface for an entire sequence of tensors, rather than on a per-tensor basis.

E.g., imagine if we had a iterable of the form

```python
[autograd, autograd, tf, tf, autograd]
```

which could feasibly happen if a user is using PennyLane NumPy for constant, non-differentiable data. However, at the moment we assume `requires_grad=True` by default for PennyLane NumPy! So if we treat each element individually, we would get this:

```python
[True, True, True, True, True]
```

Instead, if we pass `interface="tf"`, then we tell this function to 'use the TF method of determining trainability, regardless of the tensor type'. As a result, we get the expected output:

```python
[False, False, True, True, False]
```